test/e2e/trafficroute/universal_multizone/traffic_route.go

```diff
@@ -0,0 +1,295 @@
+		Expect(err).ToNot(HaveOccurred())
+
+		// Cluster 1
+		remote_1 = clusters.GetCluster(Kuma3)
+		optsRemote1 = []DeployOptionsFunc{
+			WithGlobalAddress(globalCP.GetKDSServerAddress()),
+		}
+
+		err = NewClusterSetup().
+			Install(Kuma(core.Remote, optsRemote1...)).
+			Install(DemoClientUniversal(AppModeDemoClient, defaultMesh, demoClientToken, WithTransparentProxy(true))).
+			Install(IngressUniversal(defaultMesh, ingressToken)).
+			Setup(remote_1)
+		Expect(err).ToNot(HaveOccurred())
+		err = remote_1.VerifyKuma()
+		Expect(err).ToNot(HaveOccurred())
+
+		// Cluster 2
+		remote_2 = clusters.GetCluster(Kuma4)
+		optsRemote2 = []DeployOptionsFunc{
+			WithGlobalAddress(globalCP.GetKDSServerAddress()),
+		}
+
+		err = NewClusterSetup().
+			Install(Kuma(core.Remote, optsRemote2...)).
+			Install(EchoServerUniversal("dp-echo-1", defaultMesh, "echo-v1", echoServerToken,
+				WithTransparentProxy(true),
+				WithServiceVersion("v1"),
+			)).
+			Install(EchoServerUniversal("dp-echo-2", defaultMesh, "echo-v2", echoServerToken,
+				WithTransparentProxy(true),
+				WithServiceVersion("v2"),
+			)).
+			Install(EchoServerUniversal("dp-echo-3", defaultMesh, "echo-v3", echoServerToken,
+				WithTransparentProxy(true),
+				WithServiceVersion("v3"),
+			)).
+			Install(EchoServerUniversal("dp-echo-4", defaultMesh, "echo-v4", echoServerToken,
+				WithTransparentProxy(true),
+				WithServiceVersion("v4"),
+			)).
+			Install(EchoServerUniversal("dp-backend-1", defaultMesh, "backend-v1", backendToken,
+				WithServiceName("backend"),
+				WithServiceVersion("v1"),
+				WithTransparentProxy(true),
+			)).
+			Install(IngressUniversal(defaultMesh, ingressToken)).
+			Setup(remote_2)
+		Expect(err).ToNot(HaveOccurred())
+		err = remote_2.VerifyKuma()
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	E2EAfterEach(func() {
+		// remove all TrafficRoutes
+		items, err := global.GetKumactlOptions().KumactlList("traffic-routes", "default")
+		Expect(err).ToNot(HaveOccurred())
+		for _, item := range items {
+			if item == "route-all-default" {
+				continue
+			}
+			err := global.GetKumactlOptions().KumactlDelete("traffic-route", item, "default")
+			Expect(err).ToNot(HaveOccurred())
+		}
+	})
+
+	E2EAfterSuite(func() {
+		Expect(remote_1.DeleteKuma(optsRemote1...)).To(Succeed())
+		Expect(remote_1.DismissCluster()).To(Succeed())
+
+		Expect(remote_2.DeleteKuma(optsRemote2...)).To(Succeed())
+		Expect(remote_2.DismissCluster()).To(Succeed())
+
+		Expect(global.DeleteKuma(optsGlobal...)).To(Succeed())
+		Expect(global.DismissCluster()).To(Succeed())
+	})
+
+	collectResponses := func(url string, expected, notExpected []string) {
+		instances := map[string]bool{}
+		Eventually(func() bool {
+			stdout, _, err := remote_1.ExecWithRetries("", "", "demo-client",
+				"curl", "-v", "-m", "3", "--fail", url)
+			if err != nil {
+				return false
+			}
+
+			for _, ne := range notExpected {
+				Expect(stdout).ToNot(ContainSubstring(ne))
```

Comment:
is it safe to do expect from eventually? I was under impression this is not a good idea. Will this allow Eventually to do next try or will this fail the test?